### PR TITLE
Faster get_turf() courtesy /vg/

### PR DESCRIPTION
**Pull request by ChangelingRain**
https://github.com/d3athrow/vgstation13/pull/10877

this is apparently buttloads faster please check that nerd's work for me

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -724,10 +724,7 @@ Turf and target are seperate in case you want to teleport some distance from a t
 //Gets the turf this atom inhabits
 
 /proc/get_turf(atom/A)
-	if (!istype(A))
-		return
-	for(A, A && !isturf(A), A=A.loc); //semicolon is for the empty statement
-	return A
+	return get_step(A, 0)
```
Comment (xxalpha) [inline code/__HELPERS/unsorted.dm]: Just make it a macro at this point.

**Comment by xxalpha** [comment]
This seems a bit too obvious and too good to be true, so you might want to ask lummox since he himself suggested the current get_turf implementation.

**Comment by Exxion** [comment]
You can look at the /vg/ PR to see what cases I tested it in and my speed testing
But yes, making it a macro would probably be a good idea

**Comment by xxalpha** [comment]
I just find it a little bit hard to believe lummox didn't think of this.

**Comment by Exxion** [comment]
I do too, honestly
`get_step()` is known to work with nested objects, though, and all this is doing is telling it to get the step in no direction
And I did test it with items nested as far as it is possible under normal circumstances to nest them
I only tested the speed with up to four layers of nesting, but it was the same for the first four layers, at least

**Comment by RemieRichards** [comment]
Yep, this is objectively better, I performed the same tests as when we moved from the old ```get_turf()``` to lummox's, but with lummox's as the base and this as the new one, it's just better overall and survive's MSO's "funtime runtime" snippit without runtiming (which lummox's did too)

**Comment by Exxion** [comment]
I didn't exactly use the most precise testing methods (basically just performed each of them a million times and looked at the elapsed time), but when I called `get_step()` directly rather than from the proc, it was substantially faster
And since this is so simple now, making it a macro would be easy
I asked on the BYOND forums to make sure this is intended (or at least stable) behavior, and if I get a yes, I'll probably make it into a macro
So hold on for that

**Comment by RemieRichards** [comment]
@Exxion that's basically the tests I performed last time to find out that Lummox's was 61% faster than the previous, and now to find that yours is 50% faster than Lummox's (over 1 mil calls), the only extra test I did was a small snippit of code MSO was sure would cause runtimes with Lummox's get_turf(), but it doesn't because Atoms inside their own containers simply have null locs, it's handled internally.

also Exxion I've wrote a thing on your forum post just to explain a bit better :P

File: code/__HELPERS/unsorted.dm (hunk 1 above)
Comment (RemieRichards) [inline code/__HELPERS/unsorted.dm]: macro is preferred, as we don't have to explain what ```0``` does as a direction every single time.

**Comment by RemieRichards** [comment]
>ready for review

Macro, now pls.

**Comment by optimumtact** [comment]
This is easy to undo if goes badly wrong, so lets beat everyone else to the punch and pretend we were the brains of the operation the whole time

**Comment by RemieRichards** [comment]
You're playing a dangerous game citrus boy.

**Pull request merged**